lib/internal/webstreams/util.js

```diff
@@ -183,12 +183,17 @@ function enqueueValueWithSize(controller, value, size) {
 function ensureIsPromise(fn, thisArg, ...args) {
   try {
     const value = FunctionPrototypeCall(fn, thisArg, ...args);
-    return isPromise(value) ? value : PromiseResolve(value);
+    return new Promise((r) => r(value));
   } catch (error) {
     return PromiseReject(error);
   }
 }
```

Comment:
if it's always making a new promise, any reason not to do this?
```suggestion
async function ensureIsPromise(fn, thisArg, ...args) {
  return await FunctionPrototypeCall(fn, thisArg, ...args);
}
```

Reply:
I like that a lot! An `async` function always returns a fresh `Promise`, so this should satisfy Web IDL. 🙂

Reply:
Unfortunately, this change causes some tests to fail again... 😞

However, if you remove the `await`, it *does* work:
```javascript
async function ensureIsPromise(fn, thisArg, ...args) {
  return FunctionPrototypeCall(fn, thisArg, ...args);
}
```
You still get a fresh `Promise` (from `async function`), errors are still correctly turned into rejections, and the timings match Web IDL.

Still, I don't like how sensitive these tests are to the number of microtasks. I'll try to fix this in upstream WPT.